src/scenic/core/object_types.py

```diff
@@ -1066,22 +1066,41 @@ def distanceTo(self, point):
 
     @cached_method
     def intersects(self, other):
-        """Whether or not this object intersects another object"""
-        # For objects that are boxes and flat, we can take a fast route
-        if self._isPlanarBox and other._isPlanarBox:
+        """Whether or not this object intersects another object or region"""
+        ## Type Checking ##
+        if not isinstance(other, (Object, Region)):
+            raise TypeError(
+                f"Cannot compute intersection of Scenic Object with {type(other)}."
+            )
+
+        ## Heuristic Fast Paths ##
+        # For two objects that are boxes and flat, we can take a fast route
+        if self._isPlanarBox and (isinstance(other, Object) and other._isPlanarBox):
             if abs(self.position.z - other.position.z) > (self.height + other.height) / 2:
                 return False
 
             self_poly = self._boundingPolygon
             other_poly = other._boundingPolygon
             return self_poly.intersects(other_poly)
 
-        if isLazy(self.occupiedSpace) or isLazy(other.occupiedSpace):
+        # For an object that is a box and flat with a polygonal region, we can
+        # also take a fast route.
+        if self._isPlanarBox and (isinstance(other, PolygonalRegion) and other.z == 0):
```

Comment:
Shouldn't the `other.z == 0` be something like `abs(self.position.z - other.z) <= self.height/2`?

Reply:
Hmm, actually I don't think we need to check z at all here, since 2D regions are infinite over the z axis. Does that sound right?

Reply:
Actually, those are the footprints. Should intersection use footprint?

Reply:
As we discussed, fixed to your solution for now.  For reference, we plan to discuss adding an operator that assumes footprint.